backend/src/test/java/org/cryptomator/hub/api/VaultResourceTest.java

```diff
@@ -683,6 +652,62 @@ public void testGet(String method, String path) {
 
 	}
 
+	@Nested
+	@DisplayName("/vaults/all")
+	public class GetAllVaults {
+
+		@Test
+		@DisplayName("GET /vaults/all returns 403 as user")
+		@TestSecurity(user = "User Name 1", roles = {"user"})
+		@OidcSecurity(claims = {
+				@Claim(key = "sub", value = "user1")
+		})
+		public void testGetAllVaultsAsUser() {
+			when().get("/vaults/all")
+					.then().statusCode(403);
+		}
+
+		@Test
+		@DisplayName("GET /vaults/all returns 200 as user")
+		@TestSecurity(user = "User Name 1", roles = {"admin"})
+		@OidcSecurity(claims = {
+				@Claim(key = "sub", value = "user1")
+		})
+		public void testGetAllVaultsAsAdmin() {
+			when().get("/vaults/all")
+					.then().statusCode(200)
+					.body("id", hasItems(equalToIgnoringCase("7E57C0DE-0000-4000-8000-000100001111"), equalToIgnoringCase("7E57C0DE-0000-4000-8000-000100002222"), equalToIgnoringCase("7E57C0DE-0000-4000-8000-AAAAAAAAAAAA")));
+		}
+	}
+
+	@Nested
+	@DisplayName("GET /vaults/{vaultid}/keys/{deviceId}")
+	public class Unlock {
+
+		@Test
+		@DisplayName("GET /vaults/{vaultid}/keys/{deviceId} returns 404 for not-existing vaults")
+		@TestSecurity(user = "User Name 1", roles = {"user", "admin"})
+		@OidcSecurity(claims = {
+				@Claim(key = "sub", value = "user1")
+		})
+		public void testUnlockNotExistingVault() {
+			when().get("/vaults/{vaultId}/keys/{deviceId}", "7E57C0DE-0000-4000-8000-FFFFFFFFFFFF", "someDevice")
```

Comment:
use BAD ids for non-existing or otherwise broken/invalid test data

```suggestion
			when().get("/vaults/{vaultId}/keys/{deviceId}", "7E57C0DE-0000-4000-8000-BADBADBADBAD", "someDevice")
```

Reply:
Fixed in https://github.com/cryptomator/hub/pull/202/commits/1331ecc9215fbfaa25d9797188c9c3bd72db072d